hub/src/server/config-schema.json

```diff
@@ -0,0 +1,88 @@
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "type": "object",
+  "properties": {
+    "serverName": { "type": "string" },
+    "port": { "type": "integer" },
+    "driver": { "type": { "enum":
+                          [ "aws", "azure", "disk", "google-cloud" ] } },
```

Comment:
Is there a way this list of drivers can be DRY'ed up?  There's other code (like the server initialization code) that selects which module to load based on a driver name.

Reply:
DRY'ed ?

Reply:
DRY == Don't Repeat Yourself.

Reply:
On that note, we might want to consider moving the driver-specific aspects of this schema to the driver model.

Reply:
Well, most of the point of providing this file is to have a schema file that we can link to or explicitly include in documentation -- for that, having a whole schema file is important.

We _could_ make this file auto-generate on build and commit the result.

Reply:
Ah, I see.  I'm fine with this for now as long as the number of drivers is small, but once we exceed ~10 of them we might want to reconsider auto-generating the schema.